### PR TITLE
chore: add integration tests for query builder

### PR DESCRIPTION
**Pull request by rasitozcan**
[CPT-1023](https://toptal-core.atlassian.net/browse/CPT-1023)

### Description

Add component tests for the query builder.

### How to test

- [Temploy](https://picasso.toptal.net/add-integration-tests-query-builder)
- CI should pass.

### Development checks

- [N/A] Add changeset according to [guidelines](https://github.com/toptal/picasso/blob/master/docs/contribution/changeset-guidelines.md) (if needed)
- [x] Read [CONTRIBUTING.md](https://github.com/toptal/picasso/blob/master/CONTRIBUTING.md) and [Component API principles](https://github.com/toptal/picasso/blob/master/docs/contribution/component-api.md)
- [N/A] Make sure that additions and changes on the design follow [Toptal's BASE design](https://design.toptal.net/), and it's been already discussed with designers at #-base-core
- [N/A] Annotate all `props` in component with documentation
- [N/A] Create `examples` for component
- [N/A] Ensure that deployed demo has expected results and good examples
- [N/A] Ensure the changed/created components have not caused accessibility issues. [How to use accessibility plugin in storybook](https://github.com/toptal/picasso/blob/master/docs/contribution/accessibility.md).
- [x] Self reviewed
- [x] Covered with tests ([visual tests](https://github.com/toptal/picasso/blob/master/docs/contribution/visual-testing.md) included)

**Breaking change**

- [N/A] codemod is created and showcased in the changeset
- [N/A] test alpha package of Picasso in StaffPortal

> All **_development checks_** should be done and set checked to pass the
> **GitHub Bot: TODOLess** action

<details>
<summary>PR commands</summary>
<br />

List of available commands:

- `@toptal-bot run package:alpha-release` - Release alpha version
- `@toptal-anvil ping reviewers` - Ping FX team for review

</details>

<details>
<summary>PR Review Guidelines</summary>
<br />

#### When to approve? ✅

**You are OK** with merging this PR and

1. You have no extra requests.
2. You have optional requests.
   1. Add `nit:` to your comment. (ex. `nit: I'd rename this variable from makeCircle to getCircle`)

#### When to request changes? ❌

**You are not OK** with merging this PR because

1. Something is broken after the changes.
2. Acceptance criteria is not reached.
3. Code is dirty.

#### When to comment (neither ✅ nor ❌)

**You want your comments to be addressed** before merging this PR in cases like:

1. There are leftovers like unnecessary logs, comments, etc.
2. You have an opinionated comment regarding the code that requires a discussion.
3. You have questions.

#### How to handle the comments?

1. An owner of a comment is the only one who can resolve it.
2. An owner of a comment must resolve it when it's addressed.
3. A PR owner must reply with ✅ when a comment is addressed.

</details>

[CPT-1023]: https://toptal-core.atlassian.net/browse/CPT-1023?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Comment by rasitozcan** [comment]
@toptal-anvil ping reviewers

File: cypress/component/QueryBuilder/QueryBuilder.spec.tsx [hunk 1]
```diff
@@ -0,0 +1,188 @@
...
+import type {
+  RuleGroupTypeAny,
+  RuleType,
+  ValidationResult,
+} from '@toptal/picasso-query-builder'
+import { Container } from '@toptal/picasso'
+import { useNotifications } from '@toptal/picasso/utils'
+
+const submitSuccessMessage = 'Successfully submitted'
+const firstnameValidationMessage = 'Firstname is required'
```
Comment (sashuk) [inline cypress/component/QueryBuilder/QueryBuilder.spec.tsx:12]: NIT: https://english.stackexchange.com/questions/78894/firstname-or-first-name

```suggestion
const firstNameValidationMessage = 'First name is required'
```

File: packages/picasso-query-builder/src/AddGroupButton/AddGroupButton.tsx [hunk 2]
```diff
@@ -16,6 +16,7 @@ export const AddGroupButton = ({
   disabled,
   context,
   level,
+  testID,
```
Comment (sashuk) [inline packages/picasso-query-builder/src/AddGroupButton/AddGroupButton.tsx:19]:
```suggestion
  testId,
```

Comment (rasitozcan) [reply packages/picasso-query-builder/src/AddGroupButton/AddGroupButton.tsx:19]: We have used `testIds` for the prop we use for getting testids from users. But that `testID` is some internal prop from react-querybuilder itself. It is basically a list a static test ids (like: add-rule, remove-group etc.) I have used that testID to pass to picasso elements directly. 

I may ignore `testID` completely and use `testIds` prop that users provide. wdyt?

Comment (sashuk) [reply packages/picasso-query-builder/src/AddGroupButton/AddGroupButton.tsx:19]: > I may ignore testID completely and use testIds prop that users provide. wdyt?

Sounds great, thank you!

File: packages/picasso-query-builder/src/MultiSelect/MultiSelect.tsx [hunk 3]
```diff
@@ -39,6 +40,8 @@ export const MultiSelect = ({
     return value?.split(',').filter(Boolean)
   }, [value])
 
+  console.log('test multiselect', testID)
```
Comment (sashuk) [inline packages/picasso-query-builder/src/MultiSelect/MultiSelect.tsx:43]: Some leftovers, I believe

**Comment by rasitozcan** [comment]
> @rasitozcan What do you think about adding `happoScreenshot()` to tests to make sure that visual part is also tested?

@sashuk , I can add to some. Does it play well with alerts, notifications etc, because of animations etc?

**Comment by sashuk** [comment]
@rasitozcan Great, thank you! 

> Does it play well with alerts, notifications etc, because of animations etc?

It should play well, as long as you wait for the alert content to appear on the page and then make a screenshot (and you already do it by calling `cy.getByRole('alert').should('be.visible')...`)

**Review by sashuk — APPROVED** [review]
Thank you for addressing all comments (the visual additions were approved)!

**Review by sashuk — APPROVED** [review]

**Review by angelinastavniiciuc — APPROVED** [review]

**Pull request merged**